assets/scss/components/_search.scss

```diff
@@ -86,3 +86,13 @@
         display: block;
     }
 }
+
+.c-search__icon--hide-filters {
+    fill: transparent;
+
+    .c-search__hide-filters:hover,
```

Comment:
`__hide-filters` doesn't seem to be a proper name for the element in BEM, so maybe use it rather as modifier?
`.c-search--hidden-filters` ?